katdal/dataset.py

```diff
@@ -168,6 +168,17 @@ def _selection_to_list(names, **groups):
         return [names]
 
 
+def _is_deselection(selectors):
+    """If all the selectors have a tilde ~ , then this is treated as a
+     deselect and we are going to invert the selection.
+     TODO: For version 1 release the deselector interface should just have a leading ~
```

Comment:
Part of me says that we should fix this now, because users have not used multiple `~`s before and this PR will make them used to it. Their code will then break come katdal 1.0.

On the other hand, many other things will probably break with katdal 1.0 😄

Reply:
Yeah, I think targeting 1.0 will be the best, IIRC the other deselects are on the individual terms.

Reply:
That might be true, but it's also broken...

Reply:
That's my point: no-one has ever used multiple tildes, or they would have hit a lovely "and vs or" bug.